File: java/src/main/java/frc/robot/commands/DriverCommand.java
package frc.robot.commands;

import edu.wpi.first.wpilibj.Joystick;
import edu.wpi.first.wpilibj2.command.CommandBase;
import frc.robot.subsystems.DriveSubsystem;


public class DriverCommand extends CommandBase {
    private final DriveSubsystem driveSubsystem = DriveSubsystem.getInstance();

    Joystick left_joy;
    Joystick right_joy;

    public DriverCommand(Joystick leftstick, Joystick rightstick) {

        this.left_joy = leftstick;
        this.right_joy = rightstick;

        // each subsystem used by the command must be passed into the
        // addRequirements() method (which takes a vararg of Subsystem)
        addRequirements(this.driveSubsystem);
    }

    @Override
    public void initialize() {

    }

    @Override
    public void execute() {
        driveSubsystem.Drive(() -> right_joy.getX(), () -> right_joy.getY(), () -> left_joy.getX());
    }

    @Override
    public boolean isFinished() {
        // TODO: Make this return true when this Command no longer needs to run execute()
        return false;
    }

    @Override
    public void end(boolean interrupted) {

    }
}

File: kotlin/src/main/java/frc/robot/commands/DriverCommand.kt
package frc.robot.commands

import edu.wpi.first.wpilibj.Joystick
import edu.wpi.first.wpilibj2.command.CommandBase
import frc.robot.subsystems.DriverSubsystem
import java.util.function.DoubleSupplier

class DriverCommand(left_joystick: Joystick, right_joystick: Joystick) : CommandBase() {
    private val driverSubsystem = DriverSubsystem

    private val m_left_joy = left_joystick
    private val m_right_joy = right_joystick

    init {
        // each subsystem used by the command must be passed into the addRequirements() method
        addRequirements(driverSubsystem)
    }

    override fun initialize() {}

    override fun execute() {

        driverSubsystem.drive({ m_right_joy.x }, { m_right_joy.y }, { m_left_joy.x } )
    }

    override fun isFinished(): Boolean {
        // TODO: Make this return true when this Command no longer needs to run execute()
        return false
    }



    override fun end(interrupted: Boolean) {}
}
